### PR TITLE
Fixed #29403 -- Made PyLibMCCache backend handle TooBig exception from pylibmc.

### PR DESCRIPTION
**Pull request by hramezani**
Ticket: https://code.djangoproject.com/ticket/29403
Continue to https://github.com/django/django/pull/9811

I removed the [`self._cache.delete(key)`](https://github.com/django/django/pull/9811/files#diff-d603fa6afef1a44825847ccd4e9683a6R189) and it seems we don't need it. if I add a test in `PyLibMCCacheTests` like:
```python
    def test_large_value(self):
        cache.set('key', 'a')
        large_value = 'a' * (1024 * 1024)
        cache.set('key', large_value)
        value = cache.get('key')
        self.assertIsNone(value)
```
It will be passed successfully. Do we need to add this test?

### CODE REVIEW

File: django/core/cache/backends/memcached.py [hunk 1]
```diff
@@ -213,3 +213,14 @@ def close(self, **kwargs):
...
+
+    def set(self, key, value, timeout=DEFAULT_TIMEOUT, version=None):
+        """
+        PyLibMC throws an exception if the stored value is too large,
+        python-memcache instead returns False. We need to override set
+        here to account for the difference.
+        """
+        try:
+            super().set(key, value, timeout=timeout, version=version)
+        except self._lib.TooBig:
```
Comment (felixxm) [inline django/core/cache/backends/memcached.py:225]: This breaks support for `pylibmc < 1.5.2` we should document this change in release notes and tests requirements.

Comment (hramezani) [reply django/core/cache/backends/memcached.py:225]: Thanks @felixxm, 
> we should document this change in release notes

Should we mention the new behavior(Handling the TooBig exception) or mention the `pylibmc` version update?

about the `tests requirements`, Do you mean to change `pylibmc; sys.platform != 'win32'` to `pylibmc >= 1.5.2; sys.platform != 'win32'` in `tests/tequirements/py3`?

Comment (felixxm) [reply django/core/cache/backends/memcached.py:225]: > Should we mention the new behavior(Handling the TooBig exception) or mention the `pylibmc` version update?

Removing support for older ``pylibmc`` versions, e.g. ``` * Support for ``pylibmc`` < 1.5.2 is removed.```

> about the `tests requirements`, Do you mean to change `pylibmc; sys.platform != 'win32'` to `pylibmc >= 1.5.2; sys.platform != 'win32'` in `tests/tequirements/py3`?

Yes

**Comment by felixxm** [comment]
Thanks for your efforts! However I'm going to close this per [comment](https://code.djangoproject.com/ticket/28342#comment:8).

**Pull request closed without merging**